docs/concepts/bitswap.md

```diff
@@ -11,4 +11,55 @@ related:
 
 # Bitswap
 
-<ContentStatus />
+Bitswap is a core module of IPFS for exchanging blocks of data. It directs the requesting and sending of blocks to and from other peers in the network. It is a message based protocol, wherein all messages contain wantlists or blocks. Bitswap has a [go implementation](https://github.com/ipfs/go-bitswap) and a [JavaScript implementation](https://github.com/ipfs/js-ipfs-bitswap).
+
+Bitswap has two main jobs:
+- to acquire blocks requested by the client from the network
+- to send blocks in its possession to other peers who want them
+
+### Wantlist
+
+A wantlist is a data structure that an IPFS node sends to the network to tell peers which blocks it wants. The wantlist contains [CIDs](https://docs-beta.ipfs.io/concepts/content-addressing/) (content-addressed identifiers that refer to particular blocks desired) along with priority and cancellation information.
+
+### Message
+
+A single Bitswap message may contain any of the following content:
+
+-  The sender’s wantlist. This wantlist may either be the sender’s complete wantlist or just the changes to the sender’s wantlist that the receiver needs to know.
+-  Data blocks. These are blocks that have been requested (i.e., blocks in the sender’s wantlist, as far as the receiver is aware at the time of message receipt).
+
+### Session
+
+IPFS content blocks are often connected to each other through a [MerkleDAG](https://docs-beta.ipfs.io/concepts/merkle-dag/). If a sender's block requests *are* related, a mechanism called a Bitswap session can optimize block requests sent to other peers. This can increase transfer speed and reduce the number of duplicate blocks on the network. Active peers are favored, as they are tracked relative to them having the requested blocks and how quickly they respond.
```

Comment:
```suggestion
IPFS content blocks are often connected to each other through a [Merkle DAG](https://docs-beta.ipfs.io/concepts/merkle-dag/). If a sender's block requests *are* related, a mechanism called a Bitswap session can optimize block requests sent to other peers. This can increase transfer speed and reduce the number of duplicate blocks on the network. Active peers are favored, as they are tracked relative to them having the requested blocks and how quickly they respond.
```

Reply:
Resolved in https://github.com/ipfs/ipfs-docs-v2/pull/92